### PR TITLE
[JS][16.proactive-messages] Update for 4.3.4

### PR DESCRIPTION
**Pull request by tdurnford**
## Proposed Changes

Updated proactive message sample to be in parity with CSharp sample.

## Testing

Bot sends proactive message upon receiving post request:

Annotation 2019-04-15 095102

Annotation 2019-04-15 090958

### CODE REVIEW

**Comment by Kaiqb** [comment]
@tdurnford - I just merged changes from Eric in the C# branch, so take a look at those and update the JS version. Thanks

**Comment by tdurnford** [comment]
@Kaiqb - the changes in the last couple of commits should reflect the changes to the @EricDahlvang made to the CSharp sample. Let me know if anything needs to be changed. Thanks.

File: samples/javascript_nodejs/16.proactive-messages/README.md [hunk 1]
```diff
@@ -18,6 +18,9 @@ if a bot requires some time to compile a response to the user's question, it may
 and allow the conversation to continue in the meantime. When the bot finishes compiling the response to the
 question, it will share that information with the user.
 
+This project has a NotifyController with an endpoint that will trigger the proactive messages to be sent to
```
Comment (EricDahlvang) [inline samples/javascript_nodejs/16.proactive-messages/README.md]: This is probably better represented in the node world with "This project has a notify endpoint..." or something other than "notifycontroller".

Comment (tdurnford) [reply samples/javascript_nodejs/16.proactive-messages/README.md]: Thanks for pointing that out.

File: samples/javascript_nodejs/16.proactive-messages/bots/proactiveBot.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+            await context.sendActivity(`You sent '${context.activity.text}'`);
+            await next();
+        });
+    }
+
+    addConversationReference(activity) {
+        const conversationReference = TurnContext.getConversationReference(activity);
+        this.conversationReferences[conversationReference.conversation.id] = conversationReference;
+    }
+}
```
Comment (johnataylor) [inline samples/javascript_nodejs/16.proactive-messages/bots/proactiveBot.js]: This looks good but can you make sure to run the linter and push another commit. it looks like the spacing would have been picked up by the linter.

Comment (tdurnford) [reply samples/javascript_nodejs/16.proactive-messages/bots/proactiveBot.js]: Just ran the linter and pushed another commit.

**Pull request merged**